### PR TITLE
Supporting logic for TFS integration with Unicorn via Rainbow

### PR DESCRIPTION
**Pull request by PetersonDave**
Includes:

1. ISourceControlSync registration through config
2. Update to constructor argument matching by name (ignore case -- bug fix)

### CODE REVIEW

File: src/Unicorn/Configuration/MicroConfiguration.cs [hunk 1]
```diff
@@ -90,9 +90,9 @@ public virtual object Activate(Type type, KeyValuePair<string, object>[] unmappe
 			{
 				var currentParam = ctorParams[parameterIndex];
 
-				if (unmappedConstructorParameters.Any(kv => kv.Key.Equals(currentParam.Name, StringComparison.Ordinal)))
```
Comment (kamsar) [inline src/Unicorn/Configuration/MicroConfiguration.cs]: What necessitated making parameter mapping case insensitive? If not specific to TFS, would prefer that it remain sensitive.

File: src/Unicorn/Configuration/XmlConfigurationProvider.cs [hunk 2]
```diff
@@ -87,6 +88,7 @@ protected virtual IConfiguration LoadConfiguration(XmlElement configuration, Xml
 				{"evaluator", RegisterExpectedConfigType<IEvaluator>},
 				{"predicate", RegisterExpectedConfigType<IPredicate>},
 				{"targetDataStore", RegisterExpectedConfigType<ITargetDataStore>},
+				{"sourceControlSync", RegisterExpectedConfigType<ISourceControlSync>},
```
Comment (kamsar) [inline src/Unicorn/Configuration/XmlConfigurationProvider.cs]: It won't be necessary to register this as an expected configuration, that's for "stuff Unicorn can't live without"

Normal registration should be enough, no?

File: src/Unicorn/Standard Config Files/Unicorn.config [hunk 3]
```diff
@@ -43,6 +43,9 @@
 			<defaults>
 				<!-- The source data store is where we read data from. Usually, this would be Sitecore. -->
 				<sourceDataStore type="Rainbow.Storage.Sc.SitecoreDataStore, Rainbow.Storage.Sc" singleInstance="true"/>
+
```
Comment (kamsar) [inline src/Unicorn/Standard Config Files/Unicorn.config]: This belongs in a config patch file that is specific to Unicorn.TFS, as opposed to in Unicorn core; my guess is that we'd end up with two NuGet packages: Unicorn.TFS (which essentially is this config file and probably a sample configuration that wires TFS), and Rainbow.Storage.TFS (the patches to Rainbow). Make sense?

File: src/Unicorn/Configuration/XmlConfigurationProvider.cs (hunk 2 above)
Comment (PetersonDave) [inline src/Unicorn/Configuration/XmlConfigurationProvider.cs]: Ok, that makes sense. How do I register it normally?

**Pull request closed without merging**